src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -648,6 +650,89 @@ class ActorIdStringGenerator : public Generator<std::string> {
     std::string _actorId;
 };
 
+/** `{^Now: {}}` */
+class NowGenerator : public Generator<bsoncxx::types::b_date> {
+public:
+    NowGenerator(const Node& node, GeneratorArgs generatorArgs) {}
+    bsoncxx::types::b_date evaluate() override {
+        return bsoncxx::types::b_date{std::chrono::system_clock::now()};
+    }
+};
+
+// The 2 formats also cover "%Y-%m-%d", timezones are not parsed,
+// %z, %Q are output only https://www.boost.org/doc/libs/1_75_0/doc/html/date_time/date_time_io.html.
+const std::locale formats[] = {
+    std::locale(std::locale::classic(),new boost::posix_time::time_input_facet("%Y-%m-%dT%H:%M:%S.%F")),
+    std::locale(std::locale::classic(),new boost::posix_time::time_input_facet("%Y-%m-%d %H:%M:%S.%F")),
+};
+const size_t formats_n = sizeof(formats)/sizeof(formats[0]);
+const static boost::posix_time::ptime epoch(boost::gregorian::date(1970,1,1));
+
+static unsigned long long parseDateAsMillis(const std::string &datetime) {
+    boost::posix_time::ptime pt;
+    auto millis =  0LL;
+
+    // Empty datetime is 0.
+    if(!datetime.empty()){
+        for(size_t i=0; i<formats_n; ++i) {
+            std::istringstream is(datetime);
```

Comment:
curlies avoid implicit conversions and most-vexing-parse:

```c++
std::istringstream is{datetime};
```

(Applies to a few other places as well - anywhere you instantiate a thing see if you can use `{}`. Parens is a hard habit to break if you're used to C++-11 and earlier.)